pkg/controller/stacks/templates/stackdefinition_controller_test.go

```diff
@@ -0,0 +1,241 @@
+*/
+
+package templates
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ghodss/yaml"
+	corev1 "k8s.io/api/core/v1"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/crossplaneio/crossplane-runtime/pkg/test/integration"
+	"github.com/crossplaneio/crossplane/apis"
+	"github.com/crossplaneio/crossplane/apis/stacks/v1alpha1"
+)
+
+const (
+	testStackDefinition = `
+---
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: StackDefinition
+
+metadata:
+  name: 'minimal-gcp'
+
+spec:
+  behaviors:
+    engine:
+      type: kustomize
+
+      kustomization:
+        namePrefix: mystuff
+        commonLabels:
+          group: minimal-gcp-stuff
+
+      overlays:
+      - apiVersion: gcp.crossplane.io/v1alpha3
+        kind: Provider
+        name: gcp-provider
+        bindings:
+          - from: "spec.credentialsSecretRef"
+            to: "spec.credentialsSecretRef"
+          - from: "spec.projectID"
+            to: "spec.projectID"
+      - apiVersion: v1
+        kind: ConfigMap
+        name: gcp-config
+        bindings:
+          - from: "spec.region"
+            to: "data.region"
+
+    source:
+      image: 'crossplane/minimal-gcp:0.0.1'
+
+    crd:
+      apiVersion: gcp.resourcepacks.crossplane.io/v1alpha1
+      kind: MinimalGCP
+
+    reconcile:
+    -
+      path: resources/minimal-gcp
+
+  title: "Minimal GCP Environment"
+  readme: ""
+  overview: "This stack provisions a private network and creates resource classes that has minimal node settings and refer to that private network."
+  overviewShort: "Start using GCP with Crossplane without needing to create your own resource classes!"
+  version: "1.0"
+  maintainers:
+  - name: "Muvaffak Onus"
+    email: "monus@upbound.io"
+  owners:
+  - name: "Muvaffak Onus"
+    email: "monus@upbound.io"
+  company: "Upbound"
+  category: "Environment Stack"
+  keywords:
+   - "easy"
+   - "resource class"
+   - "private network"
+   - "cheap"
+   - "minimal"
+  website: "httos://upbound.io"
+  source: "https://github.com/muvaf/minimal-gcp"
+  permissionScope: Cluster
+
+  dependsOn:
+    - crd: '*.cache.gcp.crossplane.io/v1beta1'
+    - crd: '*.compute.gcp.crossplane.io/v1alpha3'
+    - crd: '*.database.gcp.crossplane.io/v1beta1'
+    - crd: '*.storage.gcp.crossplane.io/v1alpha3'
+    - crd: '*.servicenetworking.gcp.crossplane.io/v1alpha3'
+    - crd: '*.gcp.crossplane.io/v1alpha3'
+  license: Apache-2.0
+`
+)
+
+func TestStackDefinitionController(t *testing.T) {
+	t.Skip("Skipping so that this test will not run in CI, until we have better support for it. For more context, see https://github.com/crossplaneio/crossplane/issues/1033")
```

Comment:
One potential idea here would be to use a shared branch/fork with @muvaf and @displague.  That way, you could iterate together and get new work quickly that you depend on, without going through a lot of process.  You could consider that quicker way of working instead of going through PRs to master as the integration point.

I definitely am excited about #1198 getting solved in the future :)